src/utils.ts

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+import LineSeries from './components/LineSeries';
+import type { PlotChildren } from './types';
+
+export function getMax(actionMax: number, stateMax: number | null | undefined) {
+  return stateMax === null || stateMax === undefined || stateMax <= actionMax
+    ? actionMax
+    : stateMax;
+}
+
+export function getMin(actionMin: number, stateMin: number | null | undefined) {
+  return stateMin === null || stateMin === undefined || stateMin >= actionMin
+    ? actionMin
+    : stateMin;
+}
+
+export function splitChildren(children: React.ReactNode[]) {
+  return React.Children.toArray(children).reduce<PlotChildren>(
```

Comment:
The implementation of this function would be much simpler with a for..of loop or by using array.forEach